### PR TITLE
[com_content] - Allow captcha when submit article

### PR DESCRIPTION
**Pull request by alikon**
on submit article
caparticle

on backend
admcap
#### Summary of Changes

allow to use captcha when submit an article
#### Testing Instructions

use submit article enabling/disabling captcha settings
similar to https://github.com/joomla-extensions/weblinks/pull/223

### CODE REVIEW

**Comment by sanek4life** [comment]
need as a parameter with which you can set how many items can be added such as "editor" (as long as the admin will not approve)

**Comment by brianteeman** [comment]
@sanek4life that is completely unrelated to this pr

File: administrator/language/en-GB/en-GB.com_content.ini [hunk 1]
```diff
@@ -45,6 +45,8 @@ COM_CONTENT_FEATURED_ORDER="Featured Articles Order"
 COM_CONTENT_FEATURED_TITLE="Articles: Featured"
 COM_CONTENT_FIELD_ARTICLETEXT_DESC="Enter the article content in the text area."
 COM_CONTENT_FIELD_ARTICLETEXT_LABEL="Article Text"
+COM_CONTENT_FIELD_CAPTCHA_DESC="Select the captcha plugin that will be used in the article submit form. You may need to enter required information for your captcha plugin in the Plugin Manager.<br />If 'Use Default' is selected, make sure a captcha plugin is selected in Global Configuration."
+COM_CONTENT_FIELD_CAPTCHA_LABEL="Allow Captcha on submit"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_content.ini]: Shouldn't that be require not allow?

Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_content.ini]: Ignore that as i see we use the word Allow in the contact form

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on c844e35ab1f30f4f7eebf89b6b232e1114b05307<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10976">issues.joomla.org/joomla-cms/10976</a>.</sub>

**Comment by nikitadhiman** [comment]
I have tested this item :white_check_mark: successfully on c844e35ab1f30f4f7eebf89b6b232e1114b05307<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10976">issues.joomla.org/joomla-cms/10976</a>.</sub>

**Comment by andrepereiradasilva** [comment]
to preserve current behaviour, IMO the default values should be `0` = do not use

**Comment by brianteeman** [comment]
Yes I just spotted that myself and was about to make the same comment

On 30 June 2016 at 10:49, andrepereiradasilva <notifications@github.com>
wrote:

> to preserve current behaviour, IMO the default values should be 0 = do
> not use
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/10976#issuecomment-229613534>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABPH8UZIxduInj0ABNms95fCIFxvNRGgks5qQ5ETgaJpZM4JB2qP>
> .
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by yvesh** [comment]
It would be nice if you could set who (ACL / Groups) is needed to complete the captcha. So that, for example editors and admins, would not need have it.

File: components/com_content/models/forms/article.xml [hunk 2]
```diff
@@ -203,6 +203,14 @@
...
 			class="inputbox"
 			size="1" />
+
+		<field
+			name="captcha"
+			type="captcha"
+			label="COM_CONTENT_CAPTCHA_LABEL"
+			description="COM_CONTENT_CAPTCHA_DESC"
+			validate="captcha"
+			namespace="article" />
```
Comment (andrepereiradasilva) [inline components/com_content/models/forms/article.xml]: cs: the closing xml should be in a new line

**Comment by alikon** [comment]
@andrepereiradasilva for CS we need to rewrite the whole file ?
can you help me about the @yvesh suggestion:

> It would be nice if you could set who (ACL / Groups) has to complete the captcha. So that, for example editors and admins, would not have it.

**Comment by brianteeman** [comment]
I would leave that for a separate PR

On 30 June 2016 at 18:43, Nicola Galgano <notifications@github.com> wrote:

> @andrepereiradasilva <https://github.com/andrepereiradasilva> for CS we
> need to rewrite the whole file ?
> can you help me about the @yvesh <https://github.com/yvesh> suggestion:
>
> It would be nice if you could set who (ACL / Groups) has to complete the
> captcha. So that, for example editors and admins, would not have it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/10976#issuecomment-229734520>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABPH8dEwSYvL0X_oMtAEEsDacXa93MT7ks5qRABMgaJpZM4JB2qP>
> .
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by alikon** [comment]
yes
p.s.
i was just pre-alerting the troops :yum:

**Comment by conconnl** [comment]
@alikon is this ready for testing.
Then I will put some time in it.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10976">issues.joomla.org/joomla-cms/10976</a>.</sub>

**Comment by alikon** [comment]
@conconnl yes

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on ef8def2bd6586727b51ed9dbdba160ba10b3620a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10976">issues.joomla.org/joomla-cms/10976</a>.</sub>

**Comment by conconnl** [comment]
I have tested this item :white_check_mark: successfully on ef8def2bd6586727b51ed9dbdba160ba10b3620a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10976">issues.joomla.org/joomla-cms/10976</a>.</sub>

**Comment by zero-24** [comment]
RTC. Thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10976">issues.joomla.org/joomla-cms/10976</a>.</sub>

**Comment by wilsonge** [comment]
Merged but with a tweak to move the captchaEnabled var out of the default.php file and into the view.html.php file. This shouldn't have been part of the override :) https://github.com/joomla/joomla-cms/commit/f9f796961fcb853f9d869d3a15816c0e52103922

**Comment by alikon** [comment]
thanks

**Comment by infograf768** [comment]
this has an undesired effect: now the save and cancel buttons are below the editor.  This forces to scroll for short articles as the textarea default height is important.
woul it be sensible to have a new parameter to let user choose the position of these?

**Pull request closed without merging**